scipy/interpolate/_interpolate.py

```diff
@@ -1520,9 +1549,10 @@ def integrate(self, a, b, extrapolate=None):
 
     def extend(self, c, x):
         k = max(self.c.shape[0], c.shape[0])
-        self.c = self._raise_degree(self.c, k - self.c.shape[0])
-        c = self._raise_degree(c, k - c.shape[0])
-        return _PPolyBase.extend(self, c, x)
+        with self._c_lock:
+            self.c = self._raise_degree(self.c, k - self.c.shape[0])
+            c = self._raise_degree(c, k - c.shape[0])
+            return _PPolyBase.extend(self, c, x)
```

Comment:
Got to admit I'm also confused by that the locking is here and not in the superclass call  `_PPolyBase.extend`. Is this a perf optimization to avoid locking twice, here and in the superclass call, or is there more to it --- meaning if there ever is a different `_PPolyBase`  subclass with `.extend`, it will need to follow the pattern here?

Reply:
This is to lock the access to `self.c` in `_raise_degree`